src/rust/engine/src/graph.rs

```diff
@@ -342,7 +339,7 @@ impl InnerGraph {
     let predicate = |_| true;
 
     for eid in self.walk(root_entries, false) {
-      let entry = self.entry_for_id(eid);
+      let entry = self.entry_for_id(eid).expect("Under graph lock.");
```

Comment:
I don't understand what "Under graph lock" here means. Could you elaborate a bit in the error message?

Reply:
Added an `unsafe_entry_for_id` method with a better error message. Will probably follow up to make that safe.